src/OpenTelemetry.Exporter.Jaeger/Implementation/JaegerActivityExtensions.cs

```diff
@@ -256,9 +256,26 @@ public static long ToEpochMicroseconds(this DateTimeOffset timestamp)
             return microseconds - UnixEpochMicroseconds;
         }
 
-        private static bool ProcessActivityTag(ref TagState state, KeyValuePair<string, string> activityTag)
+        private static bool ProcessActivityTag(ref TagState state, KeyValuePair<string, object> activityTag)
         {
-            var jaegerTag = new JaegerTag(activityTag.Key, JaegerTagType.STRING, activityTag.Value);
+            JaegerTag jaegerTag = default;
+
+            if (activityTag.Value is int intValue)
+            {
+                jaegerTag = new JaegerTag(activityTag.Key, JaegerTagType.LONG, vLong: intValue);
+            }
+            else if (activityTag.Value is bool boolVal)
+            {
+                jaegerTag = new JaegerTag(activityTag.Key, JaegerTagType.BOOL, vBool: boolVal);
+            }
+            else if (activityTag.Value is double doubleVal)
+            {
+                jaegerTag = new JaegerTag(activityTag.Key, JaegerTagType.DOUBLE, vDouble: doubleVal);
+            }
+            else if (activityTag.Value is string stringVal)
+            {
+                jaegerTag = new JaegerTag(activityTag.Key, JaegerTagType.STRING, stringVal);
+            }
```

Comment:
can you add my open qn here: what is the Value is of some other type? (eg: Datetime). Should we try to call .ToString() on it? or drop it?

Reply:
i think we have a duplicate part of code. let me change and we can review again.

Reply:
Right now, we already have a method that converts the default case to string:
```csharp
        public static JaegerTag ToJaegerTag(this KeyValuePair<string, object> attribute)
        {
            return attribute.Value switch
            {
                string s => new JaegerTag(attribute.Key, JaegerTagType.STRING, vStr: s),
                int i => new JaegerTag(attribute.Key, JaegerTagType.LONG, vLong: Convert.ToInt64(i)),
                long l => new JaegerTag(attribute.Key, JaegerTagType.LONG, vLong: l),
                float f => new JaegerTag(attribute.Key, JaegerTagType.DOUBLE, vDouble: Convert.ToDouble(f)),
                double d => new JaegerTag(attribute.Key, JaegerTagType.DOUBLE, vDouble: d),
                bool b => new JaegerTag(attribute.Key, JaegerTagType.BOOL, vBool: b),
                _ => new JaegerTag(attribute.Key, JaegerTagType.STRING, vStr: attribute.Value.ToString()),
            };
        }
```

Reply:
> can you add my open qn here: what is the Value is of some other type? (eg: Datetime). Should we try to call .ToString() on it? or drop it?

This is an interesting question. Has there been any efforts you're aware of at the spec level for this? I like being able to support `DateTime` versus dropping it, but dates are always hard, particularly handling them across languages. Does it make sense to adopt ISO 8601 as a standard format? I think in .NET that is done with `.ToString("o")`.

`TimeSpan` can be useful too. I think by default ToString formats these as HH:MM:SS. Does it make sense to just adopt seconds or milliseconds?

Reply:
I am not aware of any spec changes coming to support more types for attributes.
Span and SpanEvent has timestamps on it, so probably there is no need of supporting tag of type datetime ?

If one is using OTel API, they can only send supported types - string, int, bool, array. Not possible to send DateTime or any other type.
(Activity support string,object pair, so we need to decide if the exporters should do "ToString" on whatever custom objects are provided)

Reply:
Right now, the default is ToString, we could close this and open a new issue specific for other types. What do u think? @cijothomas @alanwest

Reply:
Merging this now - we need to comeback and decide if we are okay to do .ToString here.